### ISSUE TITLE
Nix Daemon disconnects on fresh install (MacOS)

### ISSUE THREAD

**Issue by Dandandooo** [NixOS/nix#12011]
## Describe the bug

I uninstalled Nix following the [guide](https://nix.dev/manual/nix/2.25/installation/uninstall.html) for the multi-user install. I then reinstall the multi-user installation of nix. I am running the latest MacOS 15.2 Developer beta. When I run any `nix-shell -p` command, I get the following error:
```
error: Nix daemon disconnected unexpectedly (maybe it crashed?)
```
There is no other output.

When I run `launchctl list | grep nix`, there is no output.
When I run 'nix-daemon' I get: 
```
error: cannot bind to socket at '/nix/var/nix/daemon-socket/socket': Address already in use
```

## Steps To Reproduce

1. `sh <(curl -L https://nixos.org/nix/install)`
2. restart
3. `nix-shell -p nix-info --run "nix-info -m"` or any other nix-shell command

## Expected behavior

I should run without error

## Metadata

nix-env (Nix) 2.25.3

## Additional context

I am running the latest MacOS 15.2 Developer Beta. I have leftover nix files in my system, but I performed the full uninstall procedure listed on the guide.

## Checklist

- [x] checked [latest Nix manual] \([source])
- [x] checked [open bug issues and pull requests] for possible duplicates

[latest Nix manual]: https://nixos.org/manual/nix/unstable/
[source]: https://github.com/NixOS/nix/tree/master/doc/manual/source
[open bug issues and pull requests]: https://github.com/NixOS/nix/labels/bug

---

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by abathur** [comment]
Is there a crash dump in console.app?

**Comment by Dandandooo** [comment]
Yes there is. This is the latest one (copied to plaintext instead of .ips):
[nix-crash.txt](https://github.com/user-attachments/files/18027306/nix-crash.txt)

**Comment by abathur** [comment]
Thanks--hopefully that'll light the way.

If you need to get Nix running, I'd try uninstalling and then installing some other recent versions to see if this is new trouble (hopefully not something to do with 15.2...). 

I'd start with the most recent 2.24 and 2.18 releases.

**Comment by GiyoMoon** [comment]
I just updated MacOS from `15.1.1` to `15.2` and am facing the same error
```
❯ nix config check
[PASS] PATH contains only one nix version.
[PASS] All profiles are gcroots.
error: cannot connect to socket at '/nix/var/nix/daemon-socket/socket': Connection refused
```
```
❯ nix-daemon
error: cannot bind to socket at '/nix/var/nix/daemon-socket/socket': Address already in use
```

**Comment by abathur** [comment]
With the caveat that this may actually be two separate issues (despite the socket being a point of commonality, the exact error and state of the daemon differs):

I checked in on Matrix and found at least one person (also running 2.24.10) able to affirm that they've taken the 15.2 update without seeing either of these errors.

I'm not familiar with what is going on here, but I'm taking it as a sign that this isn't a global issue with 15.2. 

Unless someone happens by who knows what's going on: In both cases, it might also help to spend a little time trawling the full device log in Console.app around the time of your invocations and around daemon startup (may need to stop and start it) to see if there's anything that smells related.

The only real hunch I've got at the moment is that sometimes people disable a permission at some point (often because they notice something's unhelpfully just listed as "sh" because it's a wrapper and think it looks suspicious) and then an OS update leads to new enforcement/consequences for that previous action. Worth checking for anything like that to toggle it back on for a test.

**Comment by GiyoMoon** [comment]
Running the update command from the nix manual fixed it!!
https://nix.dev/manual/nix/2.17/installation/upgrading
The version stayed the same (`2.24.10`).

**Comment by Dandandooo** [comment]
I tried this and it didn't fix it for me. Same error message

**Comment by cole-h** [comment]
Seems similar to https://github.com/NixOS/nix/issues/11918, which has another potential solution: https://discourse.nixos.org/t/ssl-ca-cert-error-on-macos/31171/5

**Comment by Dandandooo** [comment]
I have noticed that every time I run `ps -a | grep nix-daemon` the pid jumps by around 45 and the uptime is always zero. I think it might be the problem.

**Comment by Dandandooo** [comment]
I deleted everything nix related in my keychain manager, then I deleted my ssl certificate at `/etc/ssl/certs/ca-certificate.ca` and now it works.

**Comment by HansAndreManfredson** [comment]
I faced the same issue, but it turned out to be a classic OSI Layer 8 problem.

It appears I didn't use the --daemon argument during Nix installation.

**Comment by Xiang-Pan** [comment]
> Seems similar to [#11918](https://github.com/NixOS/nix/issues/11918), which has a comment linking to another potential solution: https://discourse.nixos.org/t/ssl-ca-cert-error-on-macos/31171/5

That solved my case, thanks!
